spec/requests/contact_lists_spec.rb

```diff
@@ -1,15 +1,40 @@
 require "rails_helper"
 
-RSpec.describe "ContactLists", type: :request do
+RSpec.describe "Contact list", type: :request do
+  subject(:contact_list) { FactoryBot.attributes_for(:contact_list) }
+
   before do
     user = FactoryBot.create(:user)
     sign_in user
   end
 
-  describe "GET /" do
-    it "returns http success" do
-      get "/contact_lists"
-      expect(response).to have_http_status(:success)
+  describe "when submitting a CSV file" do
+    it "shows a form to submit a new list" do
+      get "/contact_lists/new"
+      expect(response).to render_template(:new)
```

Comment:
It's not a good practice to test what template is rendered or what instance variables are assigned by the controllers. Rearranging templates or assigning different variables should not cause a controller test to fail if the behavior of the controller and the responses it has are the same.

https://github.com/rails/rails/issues/18950
>You can test what cookies are set, what HTTP code is returned, how the view looks, or what mutations happened to the DB, but testing the innards of the controller is just not a good idea.

Reply:
You can use `assert_select` to validate that certain key DOM elements are present.